Topic0/practiceTopic0/Practice2/AFDemo.java

```diff
@@ -0,0 +1,38 @@
+package practiceTopic0.Practice2;
+
+public class AFDemo {
+	public static void main(String[] args) {
+		AbstractFactory demoFactory = FactoryProducer.getFactory("FACTORY1");
```

Comment:
You code would be much better if you had used logic names instead of numbers. Use your imagination, think about concepts and try to code based on them. 
Let's say you want to create connections: Since you don't know what kind of connection you will use (SQLConnection or maybe NonSQLConnections) you create a connection factory (abstract factory)
ConnectionFactory is the abstract factory, the FactoryProducer produces SQLConnectionFactory or NonSQLConnectionFactory both implementations of ConnectionFactory.
Then you have these objects being created
MySQLConnection extends SqlConnection
OracleConnection extends SqlConnection

MongoConnection extends NonSqlConnection
CassandraConnection extends NonSqlConnection

Note: MySQL and Oracle are as you saw SQL databases. MongoDB and Cassandra are nonSQL. It doesn't matter if you didn't know that.